File: core/src/main/java/org/stepik/plugin/projectWizard/ui/RefreshButton.java
package org.stepik.plugin.projectWizard.ui;

import com.intellij.icons.AllIcons;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.event.ActionEvent;

/**
 * @author meanmail
 */
public class RefreshButton extends JButton {
    private CourseListBox target;

    public RefreshButton() {
        setIcon(AllIcons.Actions.Refresh);
    }

    void setTarget(@NotNull CourseListBox target) {
        this.target = target;
    }

    @Override
    protected void fireActionPerformed(ActionEvent event) {
        super.fireActionPerformed(event);

        if (target != null) {
            target.refresh();
        }
    }
}

File: stepik-union/src/main/java/org/stepik/core/projectWizard/ui/RefreshButton.kt
package org.stepik.core.projectWizard.ui

import com.intellij.icons.AllIcons
import java.awt.event.ActionEvent
import javax.swing.JButton


class RefreshButton : JButton() {
    internal var target: CourseListBox? = null

    init {
        icon = AllIcons.Actions.Refresh
    }

    override fun fireActionPerformed(event: ActionEvent) {
        super.fireActionPerformed(event)
        target?.refresh()
    }
}
